### PR TITLE
Migrate to rs.js 1.0.0

### PR DESCRIPTION
**Pull request by skddc**
I think this is pretty much all we need, but it would be good to [implement the timeout config in rs.js](https://github.com/remotestorage/remotestorage.js/issues/983) before merging, because currently it has to use the default timeout, which is way too low for uploads on slow connections or with a big file size.

Edit: this needs the latest commit in the widget's noglobal branch. So be sure to delete the module from `node_modules` first.

### CODE REVIEW

**Comment by skddc** [comment]
Updated to latest rs.js, RS widget, and Ember.js (phew, that wasn't comfortable).

File: .jshintrc [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "window",
     "-Promise",
     "RemoteStorage",
-    "remoteStorage"
+    "remoteStorage",
```
Comment (galfert) [inline .jshintrc]: With the new rs build and explicit importing, we can actually remove both "RemoteStorage" and "remoteStorage" from the here.

File: app/services/remotestorage.js [hunk 2]
```diff
@@ -1,58 +1,78 @@
...
+import Service from '@ember/service';
+import RemoteStorage from 'npm:remotestoragejs';
+import Widget from 'npm:remotestorage-widget';
+import Shares from 'npm:remotestorage-module-shares';
 
-export default Ember.Service.extend({
+export default Service.extend({
 
+  rs: null, // remoteStorage instance
+  shares: null,
```
Comment (galfert) [inline app/services/remotestorage.js]: Couldn't the shares rather be an `alias` of `rs.shares` instead?

Comment (skddc) [reply app/services/remotestorage.js]: As `rs` is not an EmberObject, I don't think that works.

**Comment by skddc** [comment]
> Haven't tested it yet, but the changes look pretty nice.

What's keeping you? Just `npm i && npm start`.

**Comment by galfert** [comment]
> What's keeping you? Just npm i && npm start.

Nothing was keeping me. I just did the code review first.

Testing went fine as well. Couldn't find any problems.

File: app/services/remotestorage.js [hunk 3]
```diff
@@ -1,58 +1,78 @@
...
         'connecting': true,
         'connected': false
       });
     });
+
+    this.set('rs', remoteStorage);
+    this.set('shares', remoteStorage.shares);
+
+    let widget = new Widget(remoteStorage, {
+      redirectUri: window.location.href
```
Comment (galfert) [inline app/services/remotestorage.js]: `redirectUri` is not a valid option (yet).

Comment (skddc) [reply app/services/remotestorage.js]: Thanks. Not sure where it came from. Probably the old widget/RS.

**Pull request merged**